pkg/autodiscovery/autoconfig_test.go

```diff
@@ -384,3 +386,97 @@ func TestCheckOverride(t *testing.T) {
 	})
 	assert.Len(t, ac.resolveTemplate(tpl), 1)
 }
+
+type MockSecretDecrypt struct {
+	t         *testing.T
+	scenarios []struct {
+		expectedData   []byte
+		expectedOrigin string
+		returnedData   []byte
+		returnedError  error
+		called         int
+	}
+}
+
+func (m *MockSecretDecrypt) getDecryptFunc() func([]byte, string) ([]byte, error) {
+	return func(data []byte, origin string) ([]byte, error) {
+		for n, scenario := range m.scenarios {
+			if bytes.Compare(data, scenario.expectedData) == 0 && origin == scenario.expectedOrigin {
+				m.scenarios[n].called++
+				return scenario.returnedData, scenario.returnedError
+			}
+		}
+		m.t.Errorf("Decrypt called with unexpected arguments: data=%s, origin=%s", data, origin)
+		return nil, fmt.Errorf("Decrypt called with unexpected arguments: data=%s, origin=%s", data, origin)
+	}
+}
+
+func (m *MockSecretDecrypt) haveAllScenariosBeenCalled() bool {
+	for _, scenario := range m.scenarios {
+		if scenario.called == 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func TestSecretDecrypt(t *testing.T) {
+	/// resolveTemplate
+	ac := NewAutoConfig(scheduler.NewMetaScheduler())
+	tpl := integration.Config{
+		Name:          "cpu",
+		ADIdentifiers: []string{"redis"},
+		InitConfig:    []byte("param1: ENC[foo]"),
+		Instances: []integration.Data{
+			[]byte("param2: ENC[bar]"),
+		},
+	}
+
+	mockDecrypt := MockSecretDecrypt{
+		t: t,
+		scenarios: []struct {
+			expectedData   []byte
+			expectedOrigin string
+			returnedData   []byte
+			returnedError  error
+			called         int
+		}{
+			{
+				expectedData:   []byte{},
+				expectedOrigin: "cpu",
+				returnedData:   []byte{},
+				returnedError:  nil,
+			},
+			{
+				expectedData:   []byte("param1: ENC[foo]"),
+				expectedOrigin: "cpu",
+				returnedData:   []byte("param1: foo"),
+				returnedError:  nil,
+			},
+			{
+				expectedData:   []byte("param2: ENC[bar]"),
+				expectedOrigin: "cpu",
+				returnedData:   []byte("param2: bar"),
+				returnedError:  nil,
+			},
+		},
+	}
+
+	secretsDecrypt = mockDecrypt.getDecryptFunc()
```

Comment:
You need to reset it back to normal after the test.

Reply:
Good catch ! Thanks !
Here is it: https://github.com/DataDog/datadog-agent/pull/6914/commits/af8931d5832b05d9e55715800e41133a9f28c91b